### PR TITLE
[CR] Clear the travis ccache on PR builds

### PR DESCRIPTION
**Pull request by kevingranade**
#### Summary
SUMMARY: None

#### Purpose of change
The intent here is to have PR builds on Travis continue to pull the master ccache since we expect the hit rate on the master cache to be higher than the hit rate on the PR cache.

What happens is when there are multiple merges to master between PR builds, the source code diverges from the source present on the PR branch.  When a subsequent PR build occurs, the cache has a very low hit rate because it is using the cache from the most recent PR build instead of the most recent master build.  This change prevents the PR from pushing a PR specific cache, meaning on subsequent builds it will fetch the master cache as well.

In short, this has us suffering the miss rate for any changes from the PR instead of suffering misses from changes to the master branch, which we expect to happen more often.

#### Describe the solution
Delete the .ccache directry to prevent travis from pushing a PR-specific copy of the cache.
If that works, subsequent builds of that PR will re-fetch the latest master ccache.

#### Describe alternatives you've considered
I'm not 100% sure this is the right thing to do, but I have noticed that long-lived PR builds are trending toward 0% hit rates on ccache, making them take a very long time to build.

### CODE REVIEW

**Comment by kevingranade** [comment]
Results in:
```$ echo "${TRAVIS_PULL_REQUEST}"
30825
before_cache.2
0.04s$ if [ "${TRAVIS_PULL_REQUEST}" != "false" ]; then rm -rf /home/travis/.ccache; fi
cache.2
store build cache
0.00s2.36schange detected (content changed, file is created, or file is deleted):
/home/travis/.ccache/ccache.conf
```

I think the problem is that I'm deleting the .ccache/ccache.conf file, I need to avoid deleting that in order to not trigger the cache push.

**Comment by kevingranade** [comment]
This works now, the remaining question is whether it will help or not.

**Comment by ZhilkinSerg** [comment]
So, what shall we do now? Merging this, see if it helps and revert if it won't?

**Comment by jbytheway** [comment]
Rather than relying on the details of cache layout (by deleting everything except the config file), might it be better to backup the cache at job start and restore the backed up version at the end?

**Comment by jbytheway** [comment]
As I read through the ccache docs trying to debug #33012 I found another option:  Set the env var `CCACHE_READONLY=1` and it shouldn't write anything to the cache.  Would that work?

**Comment by kevingranade** [comment]
Can I set environment variables separately for PR builds?

**Comment by ZhilkinSerg** [comment]
> 
> 
> Can I set environment variables separately for PR builds?

https://docs.travis-ci.com/user/pull-requests

```YML
script:
   - 'if [ "$TRAVIS_PULL_REQUEST" != "false" ]; then bash ./travis/run_on_pull_requests; fi'
   - 'if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then bash ./travis/run_on_non_pull_requests; fi'
```

**Comment by jbytheway** [comment]
You can set it in `build.sh` conditionally.

**Comment by kevingranade** [comment]
Closing, build times are sort of under control and I still don't know if this would help or make things worse.
